symengine/printer.h

```diff
@@ -59,7 +59,7 @@ class Precedence : public BaseVisitor<Precedence> {
             if(x.imaginary_ == 1) {
                 precedence = PrecedenceEnum::Atom;
             } else {
-                precedence = PrecedenceEnum::Mul;
+                precedence = PrecedenceEnum::Add;
```

Comment:
It should be `PrecedenceEnum::Mul`. For example `2*I` which is a `Complex` number whose printing behaves like a `Mul`.